### PR TITLE
JS: Auto-merge concurrent commits to noms datasets

### PR DESCRIPTION
**Pull request by cmasone-attic**
This patch brings JS into line with Go's handling of concurrent
commits to separate Datasets.

Fixes issue #2524

### CODE REVIEW

File: js/noms/src/remote-batch-store.js [hunk 1]
```diff
@@ -156,11 +156,13 @@ export function makeTestingRemoteBatchStore(): BatchStore {
...
 }
 
-class TestingDelegate {
+export class TestingDelegate {
   _cs: ChunkStore;
+  preUpdateRootHook: () => Promise<void>;
 
   constructor(cs: ChunkStore) {
     this._cs = cs;
+    this.preUpdateRootHook = () => Promise.resolve();
```
Comment (cmasone-attic) [inline js/noms/src/remote-batch-store.js]: It seems like this is probably the wrong no-op function to put here, but I couldn't get anything with different syntax to work.

**Comment by cmasone-attic** [comment]
PTAL

File: js/noms/src/remote-batch-store.js (hunk 1 above)
Comment (arv) [inline js/noms/src/remote-batch-store.js]: Seems fine to me.

File: js/noms/src/remote-batch-store.js [hunk 2]
```diff
@@ -156,11 +156,13 @@ export function makeTestingRemoteBatchStore(): BatchStore {
   return new RemoteBatchStore(3, new TestingDelegate(new MemoryStore()));
 }
 
-class TestingDelegate {
+export class TestingDelegate {
```
Comment (arv) [inline js/noms/src/remote-batch-store.js]: Is this for testing? Can this be moved to some test file?

**Comment by arv** [comment]
Thanks.

LGTM

**Pull request merged**